glib-macros/src/lib.rs

```diff
@@ -85,3 +88,17 @@ pub fn gflags(attr: TokenStream, item: TokenStream) -> TokenStream {
     let gen = gflags_attribute::impl_gflags(&input, &gtype_name);
     gen.into()
 }
+
+#[proc_macro_derive(Downgrade)]
```

Comment:
Can you add some documentation here and explain what it's good for?

Reply:
Sure, I just wanted to draft a PR and receive a feedback first.

Reply:
I didn't look at the code in detail but generally looks like a good idea and I wanted to have such a thing for the newtype pattern since quite a while.